### PR TITLE
fix: Fix build for thrift v0.17

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

fix: Fix build for thrift v0.17

Archlinux already upgrades `thrift` package to v0.17, but its corresponding rust lib not been published: https://issues.apache.org/jira/projects/THRIFT/issues/THRIFT-5640?filter=allopenissues

```rust
error[E0432]: unresolved import `thrift::protocol::TSerializable`
  --> /databend/target/debug/build/common-hive-meta-store-a3775ca046a58940/out/hms_patched.rs:16:142
   |
16 | ...ssageType, TInputProtocol, TOutputProtocol, TSerializable, TSetIdentifier, TStructIdentifier, TType};
   |                                                ^^^^^^^^^^^^^ no `TSerializable` in `protocol`
```

This PR uses the git version of thrift to address that.

Fix https://github.com/datafuselabs/databend/issues/7808

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1088,7 +1088,7 @@ dependencies = [
  "semver",
  "serde",
  "serfig",
- "thrift",
+ "thrift 0.18.0",
```
Comment (PsiACE) [inline Cargo.lock:1091]: It looks like this means that there are two versions of thrift at the same time, and I think that thrift 0.18 works, but what about the other older version?

Comment (Xuanwo) [reply Cargo.lock:1091]: thrift specify the version to `v0.18` after releasing `v0.17`.

Comment (Xuanwo) [reply Cargo.lock:1091]: > but what about the other older version?

`opentelemetry-jaeger` depends on `v0.15`, we don't know if we can patch this too.

**Comment by Xuanwo** [comment]
Cc @dantengsky @sandflee Can you give this PR a test?

**Pull request closed without merging**